### ISSUE TITLE
[QUESTION] Unkown error occured while connection 

### ISSUE THREAD

**Issue by shunnmugam** [mrousavy/react-native-google-nearby-messages#25]
# Question
While trying example application it's showing
2806 : Forbidden

# Environment

```
paste the output of `react-native info` here
```

```json
"react": 16.11.0,
"react-native": 0.62.2,
"react-native-google-nearby-messages": 1.0.20
```

**Comment by LeviWilliams** [comment]
Steps to reproduce? Are you able to run the example app?

**Comment by LeviWilliams** [comment]
Returning back to this issue because I just experienced it, here's the catch error from my subscribe/publish hook: 'Error: 2806: FORBIDDEN. See: https://developers.google.com/android/reference/com/google/android/gms/nearby/messages/NearbyMessagesStatusCodes'

**Comment by mrousavy** [comment]
That's very weird, especially because the Google Nearby Messages API doesn't give us more info on anything

**Comment by LeviWilliams** [comment]
Exactly, I did more looking into the error to make sure it wasn't something wrong with my API or the ability to connect to the API and I can't find anything to indicate that. I'll keep looking into this more in my application as I debug.

**Comment by LeviWilliams** [comment]
@mrousavy Any idea on what this could be? This suddenly popped up in my app and am unable to find a fix, been doing more debugging and have been able to reproduce on several devices. Still unsure about the iOS impact as I've been stuck in iOS 14 update hell.

**Comment by mrousavy** [comment]
@LeviWilliams Unfortunately I have no idea what's causing this, and I cannot reproduce it so I have to fully rely on you guys. Are you saying you're getting that error since the iOS 14 upgrade?

**Comment by mrousavy** [comment]
Whoops, just read you're talking about android. I'll check if there are any outdated packages tomorrow morning, like the google services.

**Comment by LeviWilliams** [comment]
Perfect, it could be an Android specific thing I've just been having xCode issues so haven't been able to reproduce yet.

**Comment by LeviWilliams** [comment]
Nothing wrong on the iOS side it seems, couldn't find anything wrong. @mrousavy any updates on the packages?

**Comment by LeviWilliams** [comment]
Hey sorry to bug you man @mrousavy , I've just still been unable to use my android app in production so trying to identify issues. Thanks for your help!!

**Comment by mrousavy** [comment]
@LeviWilliams sorry, I've been kinda busy with uni. I've got an exam this week, I'll see if I can find some time to take a look at this. Maybe there has been a bug at the official Nearby Messages API from Google, since I didn't change anything.

**Comment by LeviWilliams** [comment]
Sounds good @mrousavy , seems like it could be a google bug as this seemed to not appear in any of my production apps until recently. Especially since it seems to only occur when publishing a message.

**Comment by LeviWilliams** [comment]
Ever seen this @zaptrem ?

**Comment by zaptrem** [comment]
> Ever seen this @zaptrem ?

Yes, I think. Check the documentation on permissions. Google isn't explicit enough that on Android it requires either the precise location permission to be granted or the "Nearby Messages" permission (or something like that). I just went with the precise location one. @LeviWilliams @mrousavy

**Comment by LeviWilliams** [comment]
@zaptrem Here are the permissions I have:
```
    <!-- For Nearby Messaging with Bluetooth etc -->
    <uses-permission android:name="android.permission.RECORD_AUDIO" />
    <uses-permission android:name="android.permission.BLUETOOTH" />
    <uses-permission android:name="android.permission.BLUETOOTH_ADMIN" />
    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
```

**Comment by zaptrem** [comment]
> @zaptrem Here are the permissions I have:
> 
> ```
>     
>     <uses-permission android:name="android.permission.RECORD_AUDIO" />
>     <uses-permission android:name="android.permission.BLUETOOTH" />
>     <uses-permission android:name="android.permission.BLUETOOTH_ADMIN" />
>     <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
>     <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
> ```

The permission has to be programmatically requested and granted. Having it in the manifest just allows you to request it, it doesn’t guarantee  it.

**Comment by mrousavy** [comment]
@zaptrem how do you request the permission? I'd like to implement that logic in a `requestBluetoothPermission()` func 🔥

**Comment by LeviWilliams** [comment]
@zaptrem I granted all three permissions (coarse, fine, and record audio) to test using react-native-permissions and it still doesn't change the issue in this thread. @mrousavy also granted permissions in the example app and didn't change the issue.

**Comment by joseL02** [comment]
Hi! i haved the same problem i was reolve deleting the android api key because, was two android key.

**Comment by LeviWilliams** [comment]
@joseL02 is your android app currently working? I'm still unable to fix.
